### ISSUE TITLE
Feature request: Add clear button to searchField.

### ISSUE THREAD

**Issue by lllHuber** [salim-lachdhaf/searchable_dropdown#253]
It would be great to have the option to display a clear button in the search field.

I tried to add my own clear button by passing a custom controller to the searchField, but the list of options is not reset when I call `controller.clear()` in my clear-button because the list is only filtered when an onChange event occurs (`controller.clear()` does not trigger the onChange event).

So, I was able to add a custom clear-button, which clears the searchField successfully, but the filtered list is not reset when the searchField is cleared.

**Comment by salim-lachdhaf** [comment]
i added that feature, it will be available in the next release v1.0.1 , you can use it like this (see example page):
```
searchFieldProps: TextFieldProps(
                  controller: _userEditTextController,
                  decoration: InputDecoration(
                    suffixIcon: IconButton(
                      icon: Icon(Icons.clear),
                      onPressed: () {
                        _userEditTextController.clear();
                      },
                    ),
                  ),
                ),
```

**Comment by lllHuber** [comment]
@salim-lachdhaf thank you so much, that's awesome!
